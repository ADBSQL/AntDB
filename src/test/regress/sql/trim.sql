set grammar to oracle;
set datestyle='ISO,YMD';
select trim('  aa  bb  ') from dual;
select trim('  aa  bb') from dual;
select trim(leading ' ' from '   !@#$    ') from dual;
select trim(both 'x' from 'x!@#$x') from dual;
select trim(leading 'x' from 'x!@#$x') from dual;
select trim(trailing 'x' from 'x!@#$x') from dual;
select trim(both 'x' from ' x!@#$x ') from dual;
select trim(both 'xx' from ' xx!@#$x ') from dual;
select trim(both 'xl' from 'xl!@#$l') from dual;
select trim(both 'xl' from 'xl!@#$xl') from dual;
select trim(both '' from 'xl!@#$l') from dual;
select trim(both 1 from 121) from dual;
select trim(both 1 from 121.1) from dual;
select trim(both 7 from exp(2)) from dual;
select trim(leading 0 from to_date('2016-10','yyyy-mm')) from dual;
select trim(both to_char('a') from 'abc') from dual;
select trim(both null from 'abc') from dual;
