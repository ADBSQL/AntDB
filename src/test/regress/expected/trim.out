set grammar to oracle;
set datestyle='ISO,YMD';
select trim('  aa  bb  ') from dual;
 TRIM('  AA  BB  ') 
--------------------
 aa  bb
(1 row)

select trim('  aa  bb') from dual;
 TRIM('  AA  BB') 
------------------
 aa  bb
(1 row)

select trim(leading ' ' from '   !@#$    ') from dual;
 TRIM(LEADING ' ' FROM '   !@#$    ') 
--------------------------------------
 !@#$    
(1 row)

select trim(both 'x' from 'x!@#$x') from dual;
 TRIM(BOTH 'X' FROM 'X!@#$X') 
------------------------------
 !@#$
(1 row)

select trim(leading 'x' from 'x!@#$x') from dual;
 TRIM(LEADING 'X' FROM 'X!@#$X') 
---------------------------------
 !@#$x
(1 row)

select trim(trailing 'x' from 'x!@#$x') from dual;
 TRIM(TRAILING 'X' FROM 'X!@#$X') 
----------------------------------
 x!@#$
(1 row)

select trim(both 'x' from ' x!@#$x ') from dual;
 TRIM(BOTH 'X' FROM ' X!@#$X ') 
--------------------------------
  x!@#$x 
(1 row)

select trim(both 'xx' from ' xx!@#$x ') from dual;
 TRIM(BOTH 'XX' FROM ' XX!@#$X ') 
----------------------------------
  xx!@#$x 
(1 row)

select trim(both 'xl' from 'xl!@#$l') from dual;
 TRIM(BOTH 'XL' FROM 'XL!@#$L') 
--------------------------------
 !@#$
(1 row)

select trim(both 'xl' from 'xl!@#$xl') from dual;
 TRIM(BOTH 'XL' FROM 'XL!@#$XL') 
---------------------------------
 !@#$
(1 row)

select trim(both '' from 'xl!@#$l') from dual;
 TRIM(BOTH '' FROM 'XL!@#$L') 
------------------------------
 
(1 row)

select trim(both 1 from 121) from dual;
 TRIM(BOTH 1 FROM 121) 
-----------------------
 2
(1 row)

select trim(both 1 from 121.1) from dual;
 TRIM(BOTH 1 FROM 121.1) 
-------------------------
 21.
(1 row)

select trim(both 7 from exp(2)) from dual;
 TRIM(BOTH 7 FROM EXP(2)) 
--------------------------
 .3890560989306502
(1 row)

select trim(leading 0 from to_date('2016-10','yyyy-mm')) from dual;
 TRIM(LEADING 0 FROM TO_DATE('2016-10','YYYY-MM')) 
---------------------------------------------------
 2016-10-01 00:00:00
(1 row)

select trim(both to_char('a') from 'abc') from dual;
 TRIM(BOTH TO_CHAR('A') FROM 'ABC') 
------------------------------------
 bc
(1 row)

select trim(both null from 'abc') from dual;
 TRIM(BOTH NULL FROM 'ABC') 
----------------------------
 
(1 row)

